.github/workflows/build.yml

```diff
@@ -259,3 +259,25 @@ jobs:
       - run: |
           echo "Some of the required build and test workflows have failed!"
           exit 1
+
+  notify-completed-successfully-failures:
+    if: ${{ always() && needs.completed-successfully.result == 'failure' && (github.ref_name == 'main' || startsWith(github.ref_name, 'release/')) }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+    needs:
+      - completed-successfully
+    steps:
+      - name: send-notification
+        uses: slackapi/slack-github-action@007b2c3c751a190b6f0f040e47ed024deaa72844 # v1.23.0
+        # We intentionally aren't using the following here since it's from an internal repo
+        # uses: hashicorp/cloud-gha-slack-notifier@730a033037b8e603adf99ebd3085f0fdfe75e2f4 #v1
```

Comment:
The only difference here is that we're providing `SLACK_BOT_TOKEN` as an environment variable, but I thought I'd point it out.